dumper/app/src/main/java/com/google/edwmigration/dumper/application/dumper/connector/hive/AbstractHiveConnector.java

```diff
@@ -94,22 +95,22 @@ public ThriftClientPool(@Nonnull String name, @Nonnull HiveMetastoreThriftClient
             });
         }
 
-        @Nonnull
-        public ExecutorService getExecutorService() {
-            return executorService;
-        }
+        public void execute(ThriftClientConsumer consumer) {
+            executorManager.execute(() -> {
+                consumer.accept(getThreadLocalThriftClient().get());
+                return null;
+            });
+       }
 
         @Nonnull
-        public ThreadLocal<@NonNull ? extends HiveMetastoreThriftClient> getThreadLocalThriftClient() {
+        private ThreadLocal<@NonNull ? extends HiveMetastoreThriftClient> getThreadLocalThriftClient() {
             return threadLocalThriftClient;
         }
 
         @Override
-        public void close() throws IOException {
-            LOG.debug("Closing pooled Thrift client '{}'.", name);
-            final int TIMEOUT = 30;
-            LOG.debug("Shutting down thread pool backing pooled Thrift client '{}'; will wait up to {} seconds", name, TIMEOUT);
-            MoreExecutors.shutdownAndAwaitTermination(executorService, TIMEOUT, TimeUnit.SECONDS);
+        public void close() throws Exception {
+            LOG.debug("Shutting down thread pool backing pooled Thrift client '{}'", name);
+            executorManager.close();
```

Comment:
This also needs to shut down the executor, still. ExecutorManager doesn't do that. It only guarantees that any tasks submitted to it are complete.